tuna/machine.py

```diff
@@ -283,31 +307,46 @@ def get_properties(self):
 
     return self.cpus, self.gpus
 
-  def write_file(self, contents, filename=None, is_temp=False):
+  def write_file(self, contents: bytes, filename: Union[str, bytes, 'os.PathLike[str]',\
+  None] = None, is_temp: bool = False) -> Union[str, bytes, 'os.PathLike[str]', None]:
     """
     Write a file to this machine containing contents
     """
+    cnx: Connection
+    ftp: Optional[paramiko.sftp_client.SFTPClient] = None
+
+    #t_filename: Union[str, 'os.PathLike[str]']
     if is_temp:
       assert filename is None
       _, filename = tempfile.mkstemp()
     else:
       assert filename is not None
-    if self.local_machine:  # pylint: disable=no-member ; false alarm
+
+    if self.local_machine and filename is not None:  # pylint: disable=no-member ; false alarm
```

Comment:
i dont think the 'filename is not none' part is required here. The filename is established beforehand or passed in as not none.

Reply:
ok